test/python/test_cc_decoder.py

```diff
@@ -73,38 +81,22 @@ def test_maxsat_decoder(code: cc_decoder.HexagonalColorCode) -> None:
     assert est[1] == 1
 
 
-def test_simulate(distance: int, p: float, nr_sims: int) -> None:
+def test_simulate(d3_hexcode: HexagonalColorCode, distance: int, p: float, nr_sims: int) -> None:
     """Test the general simulation function."""
-    res = simulate_error_rate(distance, p, nr_sims)
+    res = simulate_error_rate(d3_hexcode, p, nr_sims)
     assert res is not None
     assert res["distance"] == distance
     assert res["p"] == p
 
 
-def check_and_load_json(file_name: str, results_dir: str) -> dict[str, Any]:
-    """Check that the results directory contains exactly one file with the given name and load it as JSON."""
-    results_path = Path(results_dir)
-    assert results_path.exists()
-    assert results_path.is_dir()
-    assert len(list(results_path.iterdir())) == 1
-    result_file = results_path / file_name
-    assert result_file.exists()
-    assert result_file.is_file()
-    with result_file.open("r") as f:
-        result: dict[str, Any] = json.load(f)
-
-    for file in results_path.iterdir():
-        file.unlink()
-    results_path.rmdir()
-
-    return result
-
-
-def test_z3_solver(script_runner: ScriptRunner, distance: int, p: float, nr_sims: int, results_dir: str) -> None:
+def test_z3_solver(
+    script_runner: ScriptRunner, d3_hexcode: HexagonalColorCode, p: float, nr_sims: int, results_dir: str
+) -> None:
     """Test the Z3 solver."""
+    d = d3_hexcode.distance
```

Comment:
there is no real need for introducing that `d` here, right? might as well be removed. Holds for all the tests.